kratos/spatial_containers/geometrical_objects_bins.cpp

```diff
@@ -89,16 +96,35 @@ void GeometricalObjectsBins::SearchInRadius(
         for(std::size_t j = min_position[1]; j < max_position[1]; j++) {
             for(std::size_t i = min_position[0]; i < max_position[0]; i++) {
                 auto& r_cell = GetCell(i, j, k);
-                for(auto p_geometrical_object : r_cell) {
-                    candidates.insert(p_geometrical_object);
+                if (IsCellBoundingBoxInsideRadius(i, j, k, rPoint, Radius)) {
+                    current_size = rResults.size();
+                    rResults.reserve(current_size + r_cell.size());
```

Comment:
I would add directly add all of them to to results here

Reply:
Okay